org.eclipse.xtext.doc/contents/310_eclipse_support.html

```diff
@@ -656,6 +656,42 @@ <h2 id="project-wizard">Project Wizard</h2>
 
 <p>In the <code>src</code> folder of the <code>ui</code> project a <code>MyDslNewProjectWizardInitialContents</code> Xtend file will also be generated, where you can specify the initial contents that your language wizard will generate.</p>
 
+<h2 id="code-mining">Code Mining</h2>
+
+<p>Code Mining shows inline annotations in the text editor that are not part of the text itself, but derived from its contents. It can be very helpful to leverage code minings for example to show inferred types, parameter names for literals and other kind of meta information.</p>
+
+<p>Code minings come in two flavors: “header annotations” are printed in a separate line above the mined text, “inline annotations” are shown in-line. The following screenshot shows both flavors:</p>
+
+<p><img src="images/code_mining.png" alt="" /></p>
+
+<p>To enable code mining, the <code>org.eclipse.xtext.xtext.generator.ui.codemining.CodeMiningFragment</code> has to be integrated in the <code>language</code> section of the MWE2 file as follows:</p>
+
+<pre><code class="language-mwe2">fragment = org.eclipse.xtext.xtext.generator.ui.codemining.CodeMiningFragment {}
+</code></pre>
+
+<p>With execution of the generator a stub class <code>&lt;LanuageName&gt;CodeMiningProvider</code> is created in the <code>.codemining</code> sub-package of the UI plugin. Furthermore the provider is registered to the the <code>org.eclipse.ui.workbench.texteditor.codeMiningProviders</code> extension point in <code>plugin.xml</code>.</p>
```

Comment:
LanuageName   |   the the

Reply:
then this needs to be fixed on the original code as well. will have a look

Reply:
where is the the the i dont see it

Reply:
"Furthermore the provider is registered to the the"